File: packages/smart-reply/ios/Plugin/Classes/Results/SuggestRepliesResult.swift
import Foundation
import Capacitor

@objc public class SuggestRepliesResult: NSObject, Result {
    let status: String
    let suggestions: [String]

    init(status: String, suggestions: [String]) {
        self.status = status
        self.suggestions = suggestions
    }

    @objc public func toJSObject() -> AnyObject {
        var suggestionsResult = JSArray()
        for suggestion in suggestions {
            suggestionsResult.append(suggestion)
        }
        var result = JSObject()
        result["status"] = status
        result["suggestions"] = suggestionsResult
        return result as AnyObject
    }
}

File: packages/smart-reply/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/smartreply/classes/results/SuggestRepliesResult.java
package io.capawesome.capacitorjs.plugins.mlkit.smartreply.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.mlkit.smartreply.interfaces.Result;
import java.util.List;

public class SuggestRepliesResult implements Result {

    @NonNull
    private final String status;

    @NonNull
    private final List<String> suggestions;

    public SuggestRepliesResult(@NonNull String status, @NonNull List<String> suggestions) {
        this.status = status;
        this.suggestions = suggestions;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSArray suggestionsArray = new JSArray();
        for (String suggestion : suggestions) {
            suggestionsArray.put(suggestion);
        }
        JSObject result = new JSObject();
        result.put("status", status);
        result.put("suggestions", suggestionsArray);
        return result;
    }
}
